src/metabase/api/search.clj

```diff
@@ -182,21 +185,24 @@
   (str "%" s "%"))
 
 (defn- search-string-clause
-  [query searchable-columns]
+  [model query]
   (when query
-    (into [:or]
-          (for [column searchable-columns
-                token (scoring/tokenize (scoring/normalize query))]
-            [:like
-             (hsql/call :lower column)
-             (wildcard-match token)]))))
+    (let [tokens (scoring/tokenize (scoring/normalize query))]
+      (into [:or]
+            (apply concat (for [column (search-config/searchable-columns-for-model model)
+                                :let [qualified-column (hsql/qualify (model->alias model) column)]]
+                            (if (and (= column :dataset_query)
+                                     (= (mdb/db-type) :postgres))
```

Comment:
Would it make sense to extract this condition as `tokenized-datasaet-query?` or something similar?

Reply:
Not really IMO, [since the line below it](https://github.com/metabase/metabase/pull/24171/files/28f5b7a7132585ad6d7bcadd9157e59a590260bc#diff-28f13071a47208d8b6ccadf005a123d921805288421a36cfc5b8dc29917240b3R361) is specific to the combination of the dataset_query column on a Postgres DB at the moment, so extracting this out as a function wouldn't hide any information. If we add full-text search for more databases and columns it could be worth extracting out the whole if block into a multimethod call. What do you think

Reply:
I've since removed this part of the code anyway, it's not supported by Postgres versions below 9.6. It doesn't affect performance much anyway.

Reply:
I think the code might turn out easier to read (and hopefully not much more or more complicated) if you make it more general, i.e., if you assume we might introduce full text search support for other DB types too. You are right that doing that would involve more than just extracting this little piece of code. In the end it's up to you to find the balance, as we don't want to write a lot of code that will be thrown away later.